util/device_sw_utils/extract_sw_logs.py

```diff
@@ -0,0 +1,217 @@
+                    size = int(section.header['sh_size'])
+                    data = section.data()
+                    ro_contents.append((base_addr, size, data))
+                else:
+                    print("Error: {} section not found in {}".format(
+                        ro_section, elf_file))
+                    sys.exit(1)
+            addr_strings = get_addr_strings(ro_contents)
+
+            # Dump the {addr: string} data.
+            rodata = ""
+            for addr in addr_strings.keys():
+                rodata += "addr: {}\n".format(hex(addr)[2:])
+                string = cleanup_newlines(addr_strings[addr])
+                rodata += "string: {}\n".format(string)
+
+            # Parse the logs fields section to extract the logs.
+            section = elf.get_section_by_name(name=logs_fields_section)
+            if section:
+                logs_base_addr = int(section.header['sh_addr'])
+                logs_size = int(section.header['sh_size'])
+                logs_data = section.data()
+            else:
+                print("Error: {} section not found in {}".format(
+                    logs_fields_section, elf_file))
+                sys.exit(1)
+
+            # Dump the logs with fields.
+            result = ""
+            num_logs = int(logs_size / LOGS_FIELDS_SIZE)
+            for i in range(num_logs):
+                start = i * LOGS_FIELDS_SIZE
+                end = start + LOGS_FIELDS_SIZE
+                severity, file_addr, line, nargs, format_addr = struct.unpack(
+                    'IIIII', logs_data[start:end])
+                result += "addr: {}\n".format(hex(logs_base_addr + start)[2:])
+                result += "severity: {}\n".format(severity)
+                result += "file: {}\n".format(
+                    prune_filename(get_str_at_addr(file_addr, addr_strings)))
+                result += "line: {}\n".format(line)
+                result += "nargs: {}\n".format(nargs)
+                result += "format: {}\n".format(
+                    cleanup_format(get_str_at_addr(format_addr, addr_strings)))
+
+            return rodata, result
+
+    except IOError as e:
+        print("Error: failed to open elf \"{}\"\n%s".format(elf_file, str(e)))
+        sys.exit(1)
+
+
+def main():
+    parser = argparse.ArgumentParser()
+    parser.add_argument('--elf-file', '-e', required=True, help="Elf file")
+    parser.add_argument('--logs-fields-section',
+                        '-f',
+                        default=LOGS_FIELDS_SECTION,
+                        help="Elf section where log fields are written.")
+    parser.add_argument('--rodata-sections',
+                        '-r',
+                        default=[RODATA_SECTION],
+                        nargs="+",
+                        help="Elf sections with rodata.")
+    parser.add_argument('--name',
+                        '-n',
+                        choices=['rom', 'sw'],
```

Comment:
I've seen that in other places as well: I see the stuggle here, but "rom" is also "sw" -- @mcy @gkelly do you have good naming suggestions for "ROM SW" vs "Flash software"?